Firestore/Swift/Tests/Codable/FirestoreEncoderTests.swift

```diff
@@ -182,29 +182,27 @@ class FirestoreEncoderTests: XCTestCase {
     assertRoundTrip(model: model, encoded: ["date": date])
   }
 
-  // Uncomment when we decide to reenable embedded DocumentReference's
-//
-//  func testDocumentReference() {
-//    struct Model: Codable, Equatable {
-//      let doc: DocumentReference
-//    }
-//    let d = FSTTestDocRef("abc/xyz")
-//    let model = Model(doc: d)
-//    assertRoundTrip(model: model, encoded: ["doc": d])
-//  }
-//
-//  // DocumentReference is not Codable unless embedded in a Firestore object.
-//  func testDocumentReferenceEncodes() {
-//    let doc = FSTTestDocRef("abc/xyz")
-//    do {
-//      _ = try JSONEncoder().encode(doc)
-//      XCTFail("Failed to throw")
-//    } catch FirebaseFirestoreSwift.FirestoreEncodingError.encodingIsNotSupported {
-//      return
-//    } catch {
-//      XCTFail("Unrecognized error: \(error)")
-//    }
-//  }
+ func testDocumentReference() {
+   struct Model: Codable, Equatable {
+     let doc: DocumentReference
+   }
+   let d = FSTTestDocRef("abc/xyz")
+   let model = Model(doc: d)
+   assertRoundTrip(model: model, encoded: ["doc": d])
+ }
+
+ // DocumentReference is not Codable unless embedded in a Firestore object.
```

Comment:
This comment and the tests don't quite match.

DocumentReference can't be encoded by the FirestoreEncoder unless it's embedded.
DocumentReference also can't be encoded by any other encoder.

This test exercises the latter case but describes the test as the former. The way to fix is to make two tests that both fail but use different encoders.

Reply:
Done.